### ISSUE TITLE
Not able to navigate within DetailsList row using tab

### ISSUE THREAD

**Issue by mankumar66** [microsoft/fluentui#20577]
#### Actual behavior: 
Try clicking on any row item of  **DetailsList -Custom Item Columns** and press **tab** key. The focus is going out of the DetailsList and the next accessible item i.e **Submit Github Issue** button is getting highlighted. Please refer the attached video clip. 
The issue can also be replicated on https://codepen.io/manishakri123/pen/OJjwQeG, try entering on any input element and press tab, it should navigate to the next input element, but the focus is coming out from the DetailsList.
I agree the fact that it's working fine on arrow keys, but it should work seamlessly with tab key as well.

#### Expected behavior: 
Try clicking on any row item of  **DetailsList -Custom Item Columns** and press **tab** key. It should navigate to the **name** column item as its a link. 

### Documentation describing expected behavior
Please investigate this issue of accessibility as we may have different input fields or Link or any actionable HTML element as a DetailsList Column. These actionable columns should be accessible through tab

https://user-images.githubusercontent.com/55076417/141237507-ca41d4da-4ee9-4c86-a5a6-9c014f0beeef.mp4

.

**Comment by gouttierre** [comment]
@mankumar66 - Thanks for filing this issue with us and providing the necessary information. To set expectations, the developers will review this issue once capacity allows.

@spmonahan - Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by spmonahan** [comment]
Hi @mankumar66 the behavior your described for `DetailsList` is correct according to design. `DetailsList` implements the [ARIA grid role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/grid_role#keyboard_interactions) which uses arrow keys to navigate between cells rather than tab. [ARIA grid practices](https://www.w3.org/TR/wai-aria-practices-1.1/#grid) also state a grid should only take one space in the tab sequence of a page to make is easy for users to quickly bypass potentially large lists of data. A good example of why this is necessary is in the first paragraph of the [Layout Grids section](https://www.w3.org/TR/wai-aria-practices-1.1/#layoutGrid) of the ARIA practices website.
